src/hexkit/protocols/dao.py

```diff
@@ -95,16 +96,13 @@ def __init__(self, *, mapping: Mapping[str, str]):
         super().__init__(message)
 
 
-class BaseModelWithId(BaseModel):
-    """A convenience base class for DTOs that have an ID field and require automatic ID
-    generation. When the model is created, a new UUID4 is generated and assigned to the
-    ID field.
-    """
+def get_uuid4_str() -> str:
+    """Generate a UUID4 string."""
+    return str(uuid4())
+
 
-    id: str = Field(
-        default_factory=lambda: str(uuid4()),
-        description="The unique identifier of the resource.",
-    )
+# provide standardized default factory for UUID4 fields
+UUID4Field = partial(Field, default_factory=get_uuid4_str)
```

Comment:
Is there a reason why you introduce the `get_uuid4_str` function? It should work with

```python
UUID4Field = partial(Field, default_factory=uuid4)
```

Even if the Field uses a `str` type instead of the `UUID4` type.

Reply:
BSON/Native UUIDs didn't work right and we've used `str` for virtually all IDs so far, so it made sense to me to continue that way. After discussing with you offline, we decided to look into a comprehensive switch from string (JSON) format in the DB to BSON as another ticket.  
For the `UUID4Field`, I've reverted to the uuid4 function.